src/mozanalysis/sizing.py

```diff
@@ -0,0 +1,383 @@
+            called.
+
+
+    Attributes:
+        experiment_name (str): Name of the study, used in naming tables
+            related to this analysis.
+        start_date (str): e.g. '2019-01-01'. First date on which enrollment
+            events were received.
+        num_dates_enrollment (int, optional): Only include this many days
+            of enrollments. If ``None`` then use the maximum number of days
+            as determined by the metric's analysis window and
+            ``last_date_full_data``. Typically ``7n+1``, e.g. ``8``. The
+            factor '7' removes weekly seasonality, and the ``+1`` accounts
+            for the fact that enrollment typically starts a few hours
+            before UTC midnight.
+        analysis_length (int, optional): Number of days to include for analysis
+        targets_sql (str): The SQL query that was executed to create the targets
+            table; only defined after get_single_window_data method is called.
+        metrics_sql (str): The SQL query that was executed to create the final
+            results table; only defined after get_single_wondow_data method is
+            called.
+
+    """
+
+    experiment_name = attr.ib()
+    start_date = attr.ib()
+    num_dates_enrollment = attr.ib(default=None)
+    analysis_length = attr.ib(default=None)
+    targets_sql = attr.ib(default='')
+    metrics_sql = attr.ib(default='')
+
+    def get_single_window_data(
+        self,
+        bq_context,
+        metric_list,
+        last_date_full_data,
+        analysis_start_days,
+        analysis_length_days,
+        target_list=None,
+        custom_targets_query=None
+    ):
+        """Return a DataFrame containing per-client metric values.
+
+        Also store them in a permanent table in BigQuery. The name of
+        this table will be printed. Subsequent calls to this function
+        will simply read the results from this table.
+
+        Args:
+            bq_context (BigQueryContext): BigQuery configuration and client.
+            metric_list (list of mozanalysis.metric.Metric): The metrics
+                to analyze.
+            last_date_full_data (str): The most recent date for which we
+                have complete data, e.g. '2019-03-22'. If you want to ignore
+                all data collected after a certain date (e.g. when the
+                experiment recipe was deactivated), then do that here.
+            analysis_start_days (int): the start of the analysis window,
+                measured in 'days since the client enrolled'. We ignore data
+                collected outside this analysis window.
+            analysis_length_days (int): the length of the analysis window,
+                measured in days.
+            target_list (list of mozanalysis.segments.Segment): The targets
+                that define clients to be included in the analysis, based on
+                inclusion in a defined user segment.
```

Comment:
Can we add something here to explain how multiple `Segment`s will be and'd together?